databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -650,6 +898,23 @@ fn convert_to_proto_stream(
     })
 }
 
+fn convert_actuation_error_to_status(
+    actuation_error: broker::ActuationError,
+    message: String,
+) -> tonic::Status {
+    match actuation_error {
+        broker::ActuationError::NotFound => tonic::Status::not_found(message),
+        broker::ActuationError::WrongType => tonic::Status::invalid_argument(message),
+        broker::ActuationError::OutOfBounds => tonic::Status::out_of_range(message),
+        broker::ActuationError::UnsupportedType => tonic::Status::invalid_argument(message),
+        broker::ActuationError::PermissionDenied => tonic::Status::permission_denied(message),
+        broker::ActuationError::PermissionExpired => tonic::Status::permission_denied(message),
```

Comment:
@lukasmittag  - this is related to your ideas on what to return, permission denied or unauthenticated

Reply:
In my opinion just fine granularity but basically if the token has expired you're unauthenticated and need to re-authenticate if you get permission denied you do not know the cause but need to fetch and process the message as well. Unauthenticated can mean two things as well since you can be unauthenticated because you're token is not correct or the token has expired. So to conclude you would either way need to have a look at the message I guess.

Reply:
Changed PermissionExpired to return tonic::Status::unauthenticated. But basically you are right, the user has to read the message to know the exact issue :)